cluster-migration/src/main/java/com/wipro/ats/bdre/clustermigration/MigrationPreprocessor.java

```diff
@@ -0,0 +1,403 @@
+        migrationPreprocessorInfo=setOozieUtilProperties(sourceDb,destDb,sourceStgtable,table,sourceJobTrackerAddress,sourceNameNodeAddress,destNameNodeAddress,sourceHiveConnection,processId,instanceExecId,migrationPreprocessorInfo);
+
+        st.close();
+        conn.close();
+        return migrationPreprocessorInfo;
+    }
+
+    private List<String> getCurrentSourcePartitionList(Statement st, String sourceDb, String table) throws Exception {
+        ResultSet rsPartitions = st.executeQuery("show partitions " + sourceDb + "." + table);
+        List<String> sourcePartitionList = new ArrayList<>();
+        while (rsPartitions.next()) {
+            sourcePartitionList.add(rsPartitions.getString(1));
+        }
+        rsPartitions.close();
+        return sourcePartitionList;
+    }
+
+    private List<String> getLastSourcePartitionList(String processId, String table) {
+        //obtaining the partition data corresponding to the previous execution for comparison with current partition data
+        ProcessLog processLog = new ProcessLog();
+        List<ProcessLogInfo> partitionLogInfos = processLog.listLastInstanceRef(Integer.parseInt(processId), table + " partition");
+        List<String> previousPartitionList = new ArrayList<>();
+        for (ProcessLogInfo processLogInfo : partitionLogInfos) {
+            previousPartitionList.add(processLogInfo.getMessage());
+        }
+        return previousPartitionList;
+    }
+
+    private Set<String> getChangedBusinessPartitionSet(List<String> sourcePartitionList, List<String> previousPartitionList) {
+        List<String> deletedPartitionList = new ArrayList<>(previousPartitionList);
+        deletedPartitionList.removeAll(sourcePartitionList);
+        List<String> addedPartitionList = new ArrayList<>(sourcePartitionList);
+        addedPartitionList.removeAll(previousPartitionList);
+        List<String> removedBusinessPartitionList = new ArrayList<>();
+        List<String> addedBusinessPartitionList = new ArrayList<>();
+
+        for (String addedPartition : addedPartitionList) {
+            addedBusinessPartitionList.add(addedPartition.substring(0, addedPartition.lastIndexOf("/")));
+        }
+
+        for (String deletedPartition : deletedPartitionList) {
+            removedBusinessPartitionList.add(deletedPartition.substring(0, deletedPartition.lastIndexOf("/")));
+        }
+        Set<String> businessPartitionSet = new HashSet<>();
+        businessPartitionSet.addAll(removedBusinessPartitionList);
+        businessPartitionSet.addAll(addedBusinessPartitionList);
+
+        for (String editedBusinessPartition : businessPartitionSet) {
+            LOGGER.debug("editedBusinessPartition = " + editedBusinessPartition);
+        }
+        return businessPartitionSet;
+    }
+
+    private List<String> getCurrentSourceColumnList(String sourceDb, String table, String sourceHiveConnection) throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS00112)